src/operations.jl

```diff
@@ -346,6 +354,28 @@ end
 
 #-----------------------------------------------------------------------------# All operations
 # :simulate => Simulate, etc.
+
+function get_callback(o::OperationRequest)
+    optype = route2operation_type[o.route]
+    get_callback(o,optype)
+end
+
+# callback for Calibrate requests
+function get_callback(o::OperationRequest, ::Type{Calibrate})
+    function (p,lossval,ode_sol)  
```

Comment:
Doesn't this need to be wrapped in a `DiscreteCallback` (or similar) object?

Reply:
The problem is that this is an Optimization problem, so I don't think the DifferentialEquations callbacks apply to those. I followed what https://docs.sciml.ai/Optimization/stable/API/solve/ has to say about callbacks for optimization problems. There is an issue in easymodelanalysis talking about a time based callback https://github.com/SciML/EasyModelAnalysis.jl/issues/158 which we could switch to if it gets implemented, but for now all we need is a function that has that particular signature.